src/Guesser/ChainGuesser.php

```diff
@@ -56,7 +57,14 @@ public function guessType($object, $name, $classes)
             if ($guesser->supportObject($object)) {
                 if ($type === null) {
                     $type = $guesser->guessType($object, $name, $classes);
-                    continue;
+
+                    // DateTime guesser should not end up in multiple types
+                    // TODO: there should be a more generic solution for this
+                    if ($guesser instanceof DateTimeGuesser) {
```

Comment:
Yes, otherwise the type is guessed to be other types as well and ends up in being a mixed type. Since this is string originally, we must not allow the SimpleTypeGuesser to guess this. Generally, I think once object types are guessed, the guesser chain should be broken.